vm/systemSmartContracts/delegation.go

```diff
@@ -424,28 +424,22 @@ func (d *delegation) addNodes(args *vmcommon.ContractCallInput) vmcommon.ReturnC
 
 func verifyIfBLSPubKeysExist(dStatus *DelegationContractStatus, arguments [][]byte) error {
 	for _, argKey := range arguments {
-		found := false
 		for _, nodeData := range dStatus.NotStakedKeys {
 			if bytes.Equal(argKey, nodeData.BLSKey) {
-				found = true
-				break
+				return fmt.Errorf("%w, key %s already exists", vm.ErrBLSPublicKeyMismatch, hex.EncodeToString(argKey))
```

Comment:
You can also add information about where it exists, just to differentiate between this return and the one below: ...already exists in not staked keys